tests/e2e/controls.py

```diff
@@ -2717,3 +2747,252 @@ def expect_icon(self, value: PatternOrStr, *, timeout: Timeout = None) -> None:
+        # data attribute of active tab and compare with value
+        playwright_expect(
+            self.loc_container.locator('a[role="tab"].active')
+        ).to_have_attribute("data-value", value, timeout=timeout)
+
+    # TODO: Make it a single locator expectation
+    # get active content instead of assertion
+    @property
+    def loc_active_content(self) -> Locator:
+        datatab_id = self.loc_container.get_attribute("data-tabsetid")
+        return self.page.locator(
+            f"div.tab-content[data-tabsetid='{datatab_id}'] > div.tab-pane.active"
+        )
+
+    def expect_content(self, value: PatternOrStr, *, timeout: Timeout = None) -> None:
+        playwright_expect(self.loc_active_content).to_have_text(value, timeout=timeout)
+
+    def expect_nav_values(
+        self,
+        value: list[PatternOrStr],
+        *,
+        timeout: Timeout = None,
+    ) -> None:
+        _MultipleDomItems.expect_locator_values_in_list(
+            page=self.page,
+            loc_container=self.loc_container,
+            el_type="a[role='tab']",
+            arr_name="value",
+            arr=value,
+            key="data-value",
+            timeout=timeout,
+        )
+
+    def expect_nav_titles(
+        self,
+        value: list[PatternOrStr],
+        *,
+        timeout: Timeout = None,
+    ) -> None:
+        self.expect.to_have_text(value, timeout=timeout)
+
+
+class LayoutNavsetTab(_LayoutNavItemBase):
+    # *args: NavSetArg,
+    # id: Optional[str] = None,
+    # selected: Optional[str] = None,
+    # header: TagChild = None,
+    # footer: TagChild = None,
+    def __init__(self, page: Page, id: str) -> None:
+        super().__init__(
+            page,
+            id=id,
+            loc_container=f"ul#{id}.nav-tabs.shiny-tab-input",
+            loc="a[role='tab']",
+        )
+
+
+class LayoutNavItem(_InputWithContainer):
+    # *args: NavSetArg,
+    # id: Optional[str] = None,
+    # selected: Optional[str] = None,
+    # header: TagChild = None,
+    # footer: TagChild = None,
+    def __init__(self, page: Page, id: str, data_value: str) -> None:
+        super().__init__(
+            page,
+            id=id,
+            loc=f"a[role='tab'][data-value='{data_value}']",
+            loc_container=f"ul#{id}",
+        )
+
+        self._data_value: str = data_value
+
+    def click(self, *, timeout: Timeout = None) -> None:
+        self.loc.click(timeout=timeout)
+
+    def expect_active(self, active: bool, *, timeout: Timeout = None) -> None:
+        _expect_class_value(self.loc, "active", active, timeout=timeout)
+
+    def expect_content(self, value: PatternOrStr, *, timeout: Timeout = None) -> None:
+        datatab_id = self.loc_container.get_attribute("data-tabsetid", timeout=timeout)
+        playwright_expect(
+            self.page.locator(
+                f"div.tab-content[data-tabsetid='{datatab_id}'] > div.tab-pane[data-value='{self._data_value}']"
+            )
```

Comment:
Similar to above, let's turn the locator into a `@property` getter function so that this class can retrieve `self.loc_content` which does the two stage retrieval.

Reply:
```suggestion
        playwright_expect(
            self.loc_content,
```